blog/2024-02-20-grpc-performance-improvements/index.mdx

```diff
@@ -0,0 +1,173 @@
+
+</details>
+
+As you can see, the performance improvements are significant for both data ingestion and query throughput. And we hope that by now you’re yelling into the screen for us to just tell you how to use it! So, here goes:
+
+## How to take advantage of gRPC in Weaviate
+
+From your perspective, taking advantage of the new gRPC API involves making sure the right sockets can be created and using the client/server solution that supports it.
+
+### Step 1: Open Port for gRPC messages
+
+First of all, the appropriate port must be open to allow gRPC messages through. Weaviate still uses REST endpoints for various functions, and gRPC traffic is sent through a separate socket than its REST traffic. Typically, the server port is 50051, although it can vary. [Weaviate Cloud Services](https://console.weaviate.cloud), for example, uses port 443 with a different URL to provide two distinct sockets for gRPC and REST traffic.
+
+<details>
+  <summary>For Docker users</summary>
+
+If you are using Docker to instantiate Weaviate, check that the gRPC traffic is forwarded appropriately, for example as shown below.
+
+---
+version: '3.4'
+services:
+  weaviate:
+    # ...
+    ports:
+    - 8080:8080
+    - 50051:50051
+    # Note the port mapping above for port 50051
+    # Remainder of `docker-compose.yml` not shown
+...
+
+</details>
+
+### Step 2: Weaviate & client library with gRPC support
+
+Next, you need a version of Weaviate that supports gRPC, and a matching client version. The reason is that gRPC requires matching server and client protobuf files for calls, messages and so on. As a result, any updates must be coupled server-side and client-side.
+
+As a rule of thumb, we recommend that you use the latest pair of server/client versions. By building with the latest versions of Weaviate server and client, you will automatically get the performance improvements.
+
+Currently, the Python client is the only one to take full advantage of the gRPC API. But, we have additional client support in the works. Our JavaScript/TypeScript client is being developed for a major update, and others will also follow.
+
+We’ll have more on these topics in the future 😉.
+
+:::tip Check out the latest Python client library
+By the way - if you’ve missed it, the new `v4` client library is a [major update](https://weaviate.io/blog/py-client-v4-release). Not just for its gRPC use, but also the hugely improved Python API. So if you have an existing codebase, the speed will be another incentive to migrate it to the latest client (e.g. [to the `v4` client API in Python](/developers/weaviate/client-libraries/python/v3_v4_migration)), along with the improved client API and support for the shiny new features.
+:::
+
+### What about the GraphQL API?
```

Comment:
This is an H2 in the draft. Is this an H3 on purpose here?

Reply:
Kind of went back & forth in my head as to whether it's a subtopic or not. I'll move it back to H2 and make it standalone.